### PR TITLE
endpoint + controller function to update new password

### PR DESCRIPTION
**Pull request by alinabudukina**
## Purpose of this PR

Create a new endpoint to check if the current entered password is correct and update it with the new one that is hashed. I added a new folder 'services' and moved all the business logic there thus the code in controller remains pure and clean

### CODE REVIEW

File: controllers/users.js [hunk 1]
```diff
@@ -64,15 +65,58 @@ exports.modifyUser = async (req, res) => {
...
+  async function passwordValidation() {
+    try {
+      const matchPassword = await bcrypt.compare(
+        req.body.password,
+        user.password,
+      );
+      if (!matchPassword) {
+        return res.status(401).json({ error: 'Mot de passe incorrect !' });
+      }
+      return true;
```
Comment (alinabudukina) [inline controllers/users.js:93]: can we return a boolean here or passing a callback function is better?

File: controllers/users.js [hunk 2]
```diff
@@ -64,15 +65,58 @@ exports.modifyUser = async (req, res) => {
...
+    } catch (error) {
+      res.status(500).json({ error });
+    }
+
+    return undefined;
```
Comment (alinabudukina) [inline controllers/users.js:98]: do we need to return undefined here?

File: controllers/users.js [hunk 3]
```diff
@@ -64,15 +65,58 @@ exports.modifyUser = async (req, res) => {
...
+      await User.updateOne({ _id: req.params.id }, passwordToUpdate);
+      return res
+        .status(200)
+        .json({ message: 'Mot de passe a été mis a jour!' });
+    } catch (error) {
+      return res.status(400).json({ error });
+    }
+  }
+
+  return undefined;
```
Comment (alinabudukina) [inline controllers/users.js:121]: do we need to return undefined here? Or isn't it better to return an updated password?

Comment (Leereen) [inline controllers/users.js:121]:
```suggestion
  async function passwordValidation() {
      return await bcrypt.compare(
        req.body.password,
        user.password,
      );
  }

try {
  const isValidPassword = await passwordValidation();

  if (isValidPassword === true) {
    try {
      const hashedPassword = await bcrypt.hash(req.body.newPassword, 10);

      const passwordToUpdate = new User({
        _id: req.params.id,
        password: hashedPassword,
      });

      await User.updateOne({ _id: req.params.id }, passwordToUpdate);
      return res
        .status(200)
        .json({ message: 'Mot de passe a été mis a jour!' });
    }
  } else {
      return res.status(401).json({ error: 'Mot de passe incorrect !' });
      }
  } catch (error) {
   return res.status(500).json({ error });
}
```

Comment (Leereen) [reply controllers/users.js:121]: Proposition to manage the boolean (is password valid) then manage the returned response to the client.

**Review by Sabevi — APPROVED** [review]

**Pull request merged**